File: redding/countries/functions/acrwelfaregains.m
%%% Solve model

function [acrwelfgain] = acrwelfaregains(param,Ctradesh,tradesh,CL,L,nobs)

global alpha sigma theta epsilon LL;

xtic = tic();

% parameters;
alpha=param(1);
theta=param(2);
epsilon=param(3);

% domestic trade share;
dtradesh=diag(tradesh);
Cdtradesh=diag(Ctradesh);

% welfare gains;
acrwelfgain=(dtradesh./Cdtradesh).^(alpha./theta);

File: redding_py/functions/acrwelfaregains.py
import numpy as np

def acrwelfaregains(param, Ctradesh, tradesh, CL, L, nobs):
    # parameters
    alpha = param[0]
    theta = param[1]
    epsilon = param[2]

    # domestic trade share
    dtradesh = np.diag(tradesh)
    Cdtradesh = np.diag(Ctradesh)

    # welfare gains
    acrwelfgain = (dtradesh / Cdtradesh) ** (alpha / theta)

    return acrwelfgain